src/mako/pixl/processors/GD.php

```diff
@@ -559,6 +559,48 @@ public function sepia(): void
 		$this->image = $temp;
 	}
 
+	/**
+	 * {@inheritdoc}
+	 */
+	public function bitonal(): void
+	{
+		if($this->hasFilters)
+		{
+			imagefilter($this->image, IMG_FILTER_GRAYSCALE);
+			imagefilter($this->image, IMG_FILTER_CONTRAST, -2000);
+		}
+		else
+		{
+			$width  = imagesx($this->image);
+			$height = imagesy($this->image);
+
+			$temp = imagecreatetruecolor($width, $height);
+
+			// Colorize pixels
+
+			for($x = 0; $x < $width; $x++)
+			{
+				for($y = 0; $y < $height; $y++)
+				{
+					$rgb = imagecolorat($this->image, $x, $y);
+
+					if((((($rgb >> 16) & 0xFF) + (($rgb >> 8) & 0xFF) + ($rgb & 0xFF)) / 3) > 0x7F)
+					{
+						imagesetpixel($temp, $x, $y, imagecolorallocate($temp, 0xFF, 0xFF, 0xFF));
+					}
+					else
+					{
+						imagesetpixel($temp, $x, $y, imagecolorallocate($temp, 0, 0, 0));
+					}
+				}
+			}
+
+			imagedestroy($this->image);
+
+			$this->image = $temp;
```

Comment:
It looks like the original image will be overridden by the `bitonal` image.

Should we have the option about overriding previous image or generating the new `bitonal` image?

Reply:
It seems to me like this is the way all filter functions work, I based this function on the `sepia` and `greyscale` functions, but maybe they should all generate a new image? Either way, there's the `snapshot` function in case one wanted to "keep" the image at a certain point.

Reply:
@Necklaces, thank you for your reply.

Maybe we have the optional parameter to add `$to` variable for generating new filtered image path :).

Reply:
You can just specify a different name when calling the `save` method if you don't want to override the original.